chain/store/store.go

```diff
@@ -367,49 +367,32 @@ func (cs *ChainStore) UnmarkBlockAsValidated(ctx context.Context, blkid cid.Cid)
 func (cs *ChainStore) SetGenesis(ctx context.Context, b *types.BlockHeader) error {
 	ts, err := types.NewTipSet([]*types.BlockHeader{b})
 	if err != nil {
-		return err
+		return xerrors.Errorf("failed to construct genesis tipset: %w", err)
 	}
 
-	if err := cs.PutTipSet(ctx, ts); err != nil {
-		return err
-	}
-
-	return cs.metadataDs.Put(ctx, dstore.NewKey("0"), b.Cid().Bytes())
-}
-
-func (cs *ChainStore) PutTipSet(ctx context.Context, ts *types.TipSet) error {
 	if err := cs.PersistTipsets(ctx, []*types.TipSet{ts}); err != nil {
-		return xerrors.Errorf("failed to persist tipset: %w", err)
+		return xerrors.Errorf("failed to persist genesis tipset: %w", err)
 	}
 
-	expanded, err := cs.expandTipset(ctx, ts.Blocks()[0])
-	if err != nil {
-		return xerrors.Errorf("errored while expanding tipset: %w", err)
+	if err := cs.AddToTipSetTracker(ctx, b); err != nil {
```

Comment:
nit: it would be nice if this just handled writing blocks internally, but I guess that's probably not critical right now?

Reply:
Yeah, not critical I think. Also not super easy to do, because we want to be batching those writes, and this method tracks things one block at a time.